SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestPropertyReplication/SpatialTestPropertyReplicationMultiworker.cpp

```diff
@@ -0,0 +1,166 @@
+ *  - Auth server check that the replicated properties are unchanged.
+ * - Clean-up:
+ *  - ReplicatedTestActor is destroyed using the RegisterAutoDestroyActor helper function.
+ */
+
+ASpatialTestPropertyReplicationMultiworker::ASpatialTestPropertyReplicationMultiworker()
+	: Super()
+{
+	Author = "Victoria Bloom";
+	Description = TEXT("This tests that the data debug mode detects property changes on non-auth servers for basic types.");
+}
+
+void ASpatialTestPropertyReplicationMultiworker::PrepareTest()
+{
+	Super::PrepareTest();
+
+	if (HasAuthority())
+	{
+		// Expected errors generated by changing replicated properties on a non-auth server
+		AddExpectedLogError(
+			TEXT("Changed actor without authority with name ReplicatedTestActorBasic_0 of type ReplicatedTestActorBasic, property "
+				 "changed without authority was ReplicatedIntProperty!"),
+			1, true);
+		AddExpectedLogError(
+			TEXT("Changed actor without authority with name ReplicatedTestActorBasic_0 of type ReplicatedTestActorBasic, property "
+				 "changed without authority was ReplicatedFloatProperty!"),
+			1, true);
+		AddExpectedLogError(
+			TEXT("Changed actor without authority with name ReplicatedTestActorBasic_0 of type ReplicatedTestActorBasic, property "
+				 "changed without authority was bReplicatedBoolProperty!"),
+			1, true);
+		AddExpectedLogError(
+			TEXT("Changed actor without authority with name ReplicatedTestActorBasic_0 of type ReplicatedTestActorBasic, property "
+				 "changed without authority was ReplicatedStringProperty!"),
+			1, true);
+	}
+
+	AddStep(
+		TEXT("The auth server spawns one ReplicatedTestActorBasic"), FWorkerDefinition::Server(1), nullptr,
+		[this]() {
+			TestActor = GetWorld()->SpawnActor<AReplicatedTestActorBasic>(FVector(0.0f, 0.0f, 50.0f), FRotator::ZeroRotator,
+																		  FActorSpawnParameters());
+			RegisterAutoDestroyActor(TestActor);
+
+			FinishStep();
+		},
+		nullptr, 5.0f);
+
+	AddStep(
+		TEXT("All workers check that they can see exactly 1 ReplicatedTestActorBasic"), FWorkerDefinition::AllWorkers, nullptr, nullptr,
+		[this](float DeltaTime) {
+			TArray<AActor*> FoundReplicatedTestActors;
+			UGameplayStatics::GetAllActorsOfClass(GetWorld(), AReplicatedTestActorBasic::StaticClass(), FoundReplicatedTestActors);
+
+			RequireEqual_Int(FoundReplicatedTestActors.Num(), 1,
+							 TEXT("The number of AReplicatedTestActorBasic found in the world should equal 1."));
+
+			if (FoundReplicatedTestActors.Num() == 1)
+			{
+				TestActor = Cast<AReplicatedTestActorBasic>(FoundReplicatedTestActors[0]);
+				RequireTrue(IsValid(TestActor), TEXT("The TestActor must be Valid (usable : non-null and not pending kill)."));
+				FinishStep();
+			}
+		},
+		5.0f);
+
+	AddStep(
+		TEXT("The auth server changes the properties"), FWorkerDefinition::Server(1),
+		[this]() -> bool {
+			return IsValid(TestActor);
+		},
+		[this]() {
+			// Replicated properties
+			TestActor->ReplicatedIntProperty = 99;
```

Comment:
I'd prefer these magic numbers to be constants defined somewhere, either in class, private namespace, local to this function, etc?